lib/_emerge/Scheduler.py

```diff
@@ -1792,6 +1801,41 @@ def _is_work_scheduled(self):
     def _running_job_count(self):
         return self._jobs
 
+    def _can_add_job(self):
+        if not super()._can_add_job():
+            return False
+
+        if (
+            self._jobs_tmpdir_blocks_threshold is not None
+            or self._jobs_tmpdir_files_threshold is not None
+        ) and hasattr(os, "statvfs"):
+            tmpdir = first_existing(
+                os.path.join(self.settings["PORTAGE_TMPDIR"], "portage")
```

Comment:
It's possible to have a separate PORTAGE_TMPDIR per PORTAGE_CONFIGROOT, so we could possibly handle multiple ROOTs here. Obviously PORTAGE_TMPDIR overrides via package.env are also completely neglected here.

Reply:
Added a loop to cover multiple ROOTs, with deduplication of identical PORTAGE_TMPDIR values.